---
title: 'GLDS-38 Analysis Using iDEP 0.91'
author: "Hyun-seok 'Ren' Chang "
date: "`r Sys.Date()`"
output: 
  html_document:
    code_folding: hide
params: 
  nVennGroupsMax: Inf
  input_folder: "GLDS37"
  geneInfoFile: "Arabidopsis_Genesinfo/Arabidopsis_thaliana__athaliana_eg_gene_GeneInfo.csv"
  geneSetFile: "Arabidopsis_Genesinfo/Arabidopsis_thaliana__athaliana_eg_gene.db"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
knitr::opts_chunk$set(fig.width = 6, fig.height = 5, fig.align = "center")
``` 

Analysis of [GLDS-38](https://genelab-data.ndc.nasa.gov/genelab/accession/GLDS-38/) from NASA GeneLab 

This R markdown file was auto-generated by the [iDEP website](http://ge-lab.org/idep/)
Using iDEP 0.91, originally by Steven Xijin.Ge@sdstate.edu 

Ge SX, Son EW, Yao R: iDEP: an integrated web application for differential expression and pathway analysis of RNA-Seq data. BMC Bioinformatics 2018, 19(1):534. PMID:30567491	

## 1. Read data  

First we set up the working directory to where the files are saved.    

```{r, message=FALSE}
input_biclustMethod_ <- "BCCC()"
```
R packages used 

```{r, message=FALSE}
library(RSQLite, verbose = FALSE) # for database connection
library(gplots, verbose = FALSE) # for hierarchical clustering
library(ggplot2, verbose = FALSE) # graphics
library(e1071, verbose = FALSE) # computing kurtosis
#library(DT, verbose = FALSE) # for renderDataTable
library(plotly, verbose = FALSE) # for interactive heatmap
library(reshape2, verbose = FALSE) # for melt correlation matrix in heatmap

# From Data Read Function
library(edgeR, verbose = FALSE) # count data D.E.
library(DESeq2, verbose = FALSE) # count data analysis, DEG.DESeq2

# TSNE Plot, tSNEgenePlot
library(Rtsne, verbose = FALSE)

# PGSA Pathway PGSEA Pathway, PGSEAplot
library(PGSEA, verbose = FALSE)

# DEG.limma
library(limma, verbose = FALSE) # Differential expression
library(statmod, verbose = FALSE)

# enrichment plot
library(dendextend) # customizing tree

# enrich.net2, moduleNetwork
library(igraph)

# Stringdb_geneList, StringDB_GO_enrichmentData, stringDB_network1
# StringDB_network_link
library(STRINGdb, verbose = FALSE)

# gagePathwayData
library(gage, verbose = FALSE) # pathway analysis

# fgseaPathwayData
library(fgsea, verbose = FALSE) # fast GSEA

# ReactomePAPathwayData
library(ReactomePA, verbose = FALSE) # pathway analysis

# KeggImage
library(pathview)

# genomePlot, genomePlotDataPre
library(PREDA, verbose = FALSE) # showing expression on genome
library(PREDAsampledata, verbose = FALSE)
library(hgu133plus2.db, verbose = FALSE)

# biclustering
library(biclust, verbose = FALSE)

library(knitr) #  install if needed. for showing tables with kable
library(kableExtra)

if (input_biclustMethod_ == "BCQU()") {
  library(QUBIC, verbose = FALSE)
} # have trouble installing on Linux
if (input_biclustMethod_ == "BCUnibic()") {
  library(runibic, verbose = FALSE)
} # Test biclustMethod dependant qubic runibic

# wgcna
library(WGCNA)
library(flashClust, verbose = FALSE)
```

```{r, message=FALSE  }  
source("iDEP_core_functions_only.R")
```  

```{r, message=FALSE}
# Each row of this matrix represents a color scheme;
mycolors_ <- sort(rainbow(20))[c(1, 20, 10, 11, 2, 19, 3, 12, 4, 13, 5, 14, 6, 15, 7, 16, 8, 17, 9, 18)]

hmcols_ <- colorRampPalette(colors = c('#4575B4', '#91BFDB', '#E0F3F8', '#FFFFBF', '#FEE090', '#FC8D59', '#D73027'))(75)

heatColors_ <- rbind(
  greenred(75), 
  bluered(75), 
  colorpanel(75, "green", "black", "magenta"), 
  colorpanel(75, "blue", "yellow", "red"), 
  hmcols_
)

rownames(heatColors_) <- c("Green-Black-Red", "Blue-White-Red", "Green-Black-Magenta", "Blue-Yellow-Red", "Blue-white-brown")
```
We are using the downloaded gene expression file where gene IDs has 
been converted to Ensembl gene IDs. This is because the ID conversion database is too large
to download. You can use your original file if your file uses Ensembl ID, or you do not want 
to use the pathway files available in iDEP (or it is not available).  

```{r, message=FALSE  }
inputFolderFiles <- list.files(params$input_folder, full.names = TRUE)

inputFile_ <- inputFolderFiles[stringr::str_detect(tolower(inputFolderFiles), "expression.csv$")]
sampleInfoFile_ <- inputFolderFiles[stringr::str_detect(tolower(inputFolderFiles), "sampleinfo.csv$")]
gldsMetadataFile_ <- inputFolderFiles[stringr::str_detect(tolower(inputFolderFiles), "metadata.csv$")]


geneInfoFile_ <- params$geneInfoFile
geneSetFile_ <- params$geneSetFile # pathway database in SQL; can be GMT format

STRING10_speciesFile_ <- "https://raw.githubusercontent.com/iDEP-SDSU/idep/master/shinyapps/idep/STRING10_species.csv"
``` 

```{r, message=FALSE  }  
readMetadata.out_ <- readMetadata(inFile = gldsMetadataFile_) #gldsMetadataFile_)

kable(readMetadata.out_) %>%
  kable_styling(bootstrap_options = c("striped", "hover")) %>%
  scroll_box(width = "100%")
```

```{r, message=FALSE  }  
readData.out_ <- readData(inFile = inputFile_, 
                          input_missingValue = "geneMedian", 
                          input_dataFileFormat = 1, 
                          input_minCounts = 0.5, 
                          input_NminSamples = 1, 
                          input_countsLogStart = 4, 
                          input_CountsTransform = 1)

kable(head(readData.out_$data)) %>%
  kable_styling(bootstrap_options = c("striped", "hover")) %>%
  scroll_box(width = "100%")
```


```{r, message=FALSE  }  
readSampleInfo.out_ <- readSampleInfo(inFile = sampleInfoFile_, 
                                      readData.out = readData.out_)
kable(readSampleInfo.out_) %>%
  kable_styling(bootstrap_options = c("striped", "hover")) %>%
  scroll_box(width = "100%")
```


```{r, message=FALSE  }  
input_noIDConversion_ <- TRUE

allGeneInfo.out_ <- geneInfo(fileName = geneInfoFile_)
converted.out_ <- NULL
convertedData.out_ <- convertedData(converted.out = NULL, 
                                    readData.out = readData.out_, 
                                    input_noIDConversion = TRUE)

nGenesFilter(readData.out = readData.out_, 
             converted.out = NULL, 
             convertedData.out = convertedData.out_, 
             input_noIDConversion = TRUE)

convertedCounts.out_ <- convertedCounts(readData.out = readData.out_, converted.out = NULL) # converted counts, just for compatibility
```

## 2. Pre-process 

```{r, message = FALSE}  
# Read counts per library
parDefault_ <- par()
par(mar = c(12, 4, 2, 2))
# barplot of total read counts
rawCounts <- readData.out_$rawCounts
groups_ <- as.factor(detectGroups(colnames(rawCounts)))
if (nlevels(groups_) <= 1 | nlevels(groups_) > 20) {
  col1_ <- "green"
} else {
  col1_ <- rainbow(nlevels(groups_))[groups_]
}

barplot(colSums(readData.out_$rawCounts) / 1e6,
        col = col1_, las = 3, main = "Total read counts (millions)"
)
readCountsBias(readData.out = readData.out_, readSampleInfo.out = readSampleInfo.out_) # detecting bias in sequencing depth
```
```{r, message=FALSE  }  
# Box plot
boxplot(
  x = readData.out_$data,
  las = 2, col = col1_,
  ylab = "Transformed expression levels",
  main = "Distribution of transformed data"
)
```
```{r, message=FALSE  }  
# Density plot
par(parDefault_)
densityPlot(readData.out = readData.out_, 
            mycolors = mycolors_)
```
```{r, message=FALSE  }  
# Scatter plot of the first two samples
plot(
  x = readData.out_$data[, 1:2],
  xlab = colnames(readData.out_$data)[1], 
  ylab = colnames(readData.out_$data)[2],
  main = "Scatter plot of first two samples"
)
```
```{r, message=FALSE  }  
#### plot gene or gene family
genePlot(allGeneInfo.out = allGeneInfo.out_, 
         convertedData.out = convertedData.out_, 
         input_selectOrg = "BestMatch", 
         input_geneSearch = "HOXA")
```


```{r, message=FALSE  } 
geneBarPlotError(convertedData.out = convertedData.out_, 
                 allGeneInfo.out = allGeneInfo.out_, 
                 input_selectOrg = 'BestMatch', 
                 input_geneSearch = "HOXA", 
                 input_useSD = "FALSE") # Use standard deviation instead of standard error in error bar?
```

## 3. Heatmap  
```{r, message=FALSE  }  
# hierarchical clustering tree
x <- readData.out_$data
maxGene <- apply(x, 1, max)
# remove bottom 25% lowly expressed genes, which inflate the PPC
x <- x[which(maxGene > quantile(maxGene)[1]), ]
plot(as.dendrogram(hclust2(dist2(t(x)))), ylab = "1 - Pearson C.C.", type = "rectangle")
```

```{r, message=FALSE  }  
# Correlation matrix
#input_labelPCC_ <- TRUE # Show correlation coefficient?
correlationMatrix(readData.out = readData.out_, input_labelPCC = TRUE)
```
```{r, message=FALSE  }  
png(paste(params$input_folder, "heatmap.png", sep = "/"), width = 10, height = 15, units = "in", res = 300)
staticHeatmap(readData.out = readData.out_, 
              readSampleInfo.out = readSampleInfo.out_, 
              heatColors = heatColors_, 
              input_nGenes = 1000, 
              input_geneCentering = TRUE, 
              input_sampleCentering = FALSE, 
              input_geneNormalize = FALSE, 
              input_sampleNormalize = FALSE, 
              input_noSampleClustering = FALSE, 
              input_heatmapCutoff = 4, 
              input_distFunctions = 1, 
              input_hclustFunctions = 1, 
              input_heatColors1 = 1, 
              input_selectFactorsHeatmap = 'Gravity')
dev.off()
```
![heatmap] (`r paste(params$input_folder, "heatmap.png", sep = "/")`)   
```{r, message=FALSE  }  
heatmapPlotly(convertedData.out = convertedData.out_, 
              heatColors = heatColors_, 
              allGeneInfo.out = allGeneInfo.out_, 
              input_geneCentering = TRUE, 
              input_sampleCentering = FALSE, 
              input_geneNormalize = FALSE, 
              input_sampleNormalize = FALSE, 
              input_heatColors1 = 1)# interactive heatmap using Plotly
```

## 4. K-means clustering

```{r, message=FALSE  } 
distributionSD(convertedData.out = convertedData.out_, 
               input_nGenesKNN = 2000) # Distribution of standard deviations
```     

```{r, message=FALSE  }  
KmeansNclusters(convertedData.out = convertedData.out_, 
                input_nGenesKNN = 2000) # Number of clusters
```     

```{r, message=FALSE  }  
Kmeans.out_ <- Kmeans(convertedData.out = convertedData.out_,
                      maxGeneClustering = 12000, 
                      input_nGenesKNN = 2000, 
                      input_nClusters = 4, 
                      input_kmeansNormalization = "geneMean", 
                      input_KmeansReRun = 0) # Running K-means

KmeansHeatmap(Kmeans.out = Kmeans.out_, 
              .mycolors = mycolors_, 
              .heatColors = heatColors_, 
              .input_heatColors1 = 1) # Heatmap for k-Means
```     

```{r, message=FALSE  }  
# Read gene sets for enrichment analysis
GeneSets.out_ <- readGeneSets(
  fileName = geneSetFile_,
  convertedData = convertedData.out_, 
  GO = "GOBP", 
  selectOrg = "NEW",
  myrange = c(15, 2000)
)

# Alternatively, users can use their own GMT files by
# GeneSets.out_ <- readGMTRobust('somefile.GMT')
results <- KmeansGO(Kmeans.out = Kmeans.out_, 
                    input_nClusters = 4, 
                    GeneSets.out = GeneSets.out_) # Enrichment analysis for k-Means clusters

results$adj.Pval <- format(results$adj.Pval, digits = 3)

kable(results, row.names = FALSE) %>%
  kable_styling(bootstrap_options = c("striped", "hover")) %>%
  scroll_box(width = "100%")
```     


```{r, message=FALSE  } 

tSNEgenePlot(Kmeans.out_, 
             input_seedTSNE = 0, 
             input_colorGenes = TRUE, # Color genes in t-SNE plot?
             mycolors = mycolors_) # Plot genes using t-SNE
```

## 5. PCA and beyond   

```{r, message=FALSE  } 
PCAplot(convertedData.out = convertedData.out_, 
        readSampleInfo.out = readSampleInfo.out_, 
        input_selectFactors = colnames(readSampleInfo.out_)[1], 
        input_selectFactors2 = colnames(readSampleInfo.out_)[2])
```     
```{r, message=FALSE  }  
MDSplot(convertedData.out = convertedData.out_, 
        readSampleInfo.out = readSampleInfo.out_, 
        input_selectFactors = colnames(readSampleInfo.out_)[1], 
        input_selectFactors2 = colnames(readSampleInfo.out_)[2])
```     
```{r, message=FALSE  }  
tSNEplot(convertedData.out = convertedData.out_, 
         readSampleInfo.out = readSampleInfo.out_, 
         input_selectFactors = colnames(readSampleInfo.out_)[1],
         input_selectFactors2 = colnames(readSampleInfo.out_)[2],
         input_tsneSeed2 = 0)
```

```{r, message=FALSE  }  
# Read gene sets for pathway analysis using PGSEA on principal components
GeneSets.out_ <- readGeneSets(
  fileName = geneSetFile_,
  convertedData = convertedData.out_, 
  GO = "GOBP", 
  selectOrg = "NEW",
  myrange = c(15, 2000)
)

PCApathway(convertedData.out = convertedData.out_, 
           GeneSets.out = GeneSets.out_) # Run PGSEA analysis
```     
```{r, message=FALSE  }  
cat(
  PCA2factor(readData.out = readData.out_, 
             readSampleInfo.out = readSampleInfo.out_)
) # The correlation between PCs with factors
```

## 6. DEG1   


```{r message=FALSE}
# List to hold limma outputs for all variables
limma_outputs <- setNames(as.list(rep(NA, ncol(readSampleInfo.out_))),
                          nm = colnames(readSampleInfo.out_))

# List to hold all comparisons for each variable
comps <- limma_outputs 

# List to hold all DEG.out for each variable
DEG_output <- limma_outputs 

for (variable in colnames(readSampleInfo.out_)){
  values <- unique(readSampleInfo.out_[, variable])
  
  # Create all combinations for this variable  
  all_comps <- outer(values, values, function(x,y) paste0(variable, ": ", x, " vs. ", y))
  comps[[variable]] <- all_comps[upper.tri(all_comps)]
}

# Run limma for all variables. lapply runs the function for each element of the list that you provide.
limma_outputs <- lapply(comps, 
                        function(x) 
                          limma(convertedData.out = convertedData.out_, 
                                readSampleInfo.out = readSampleInfo.out_, 
                                input_dataFileFormat = 1, 
                                input_countsLogStart = 4, 
                                convertedCounts.out = convertedCounts.out_, 
                                input_CountsDEGMethod = 3, # 3 
                                input_limmaPval = 0.1, 
                                input_limmaFC = 3, 
                                input_selectModelComprions = x, 
                                input_selectFactorsModel = unlist(strsplit(x[1], split = ":"))[1],
                                input_selectInteractions = NULL, 
                                input_selectBlockFactorsModel = NULL, 
                                factorReferenceLevels.out = NULL)
)

DEG.data_out <- lapply(limma_outputs,
                       function(x) 
                         DEG.data(limma.out = x,
                                  convertedData.out = convertedData.out_, 
                                  allGeneInfo.out = allGeneInfo.out_))
```
```{r results = "asis"}  
for (var in names(limma_outputs)){
  # Print header for subsection
  cat("### Limma for", var) 
  cat("\n\n")
  
  # Print all comparisons done.
  cat("Comparisons done for this variable:\n") 
  cat(limma_outputs[[var]]$comparisons, sep = ", ")
  cat("\n")
  
  ## Venn diagram.
  nVennGroups <- min(params$nVennGroupsMax, length(limma_outputs[[var]]$comparisons)) # if less than three comparisons, include all comparisons.
  vennPlot(limma.out = limma_outputs[[var]], 
           input_selectComparisonsVenn = limma_outputs[[var]]$comparisons[1:nVennGroups], 
           input_UpDownRegulated = FALSE) # Split up and down regulated genes
  
  print(sigGeneStats(limma_outputs[[var]])) # number of DEGs as figure
  
  kable(sigGeneStatsTable(limma_outputs[[var]]), 
        row.names = FALSE) %>% # number of DEGs as table
    kable_styling(bootstrap_options = c("striped", "hover"), position = "center") %>% 
    print()
}
```


<!-- ## 7. DEG2 -->

<!-- ```{r, message=FALSE  }  -->
<!-- # input_selectContrast_ <- "Microgravity-Terrestrial" # Selected comparisons -->
<!-- selectedHeatmap.data.out_ <- selectedHeatmap.data( -->
<!--   convertedData.out = convertedData.out_,  -->
<!--   readSampleInfo.out = readSampleInfo.out_,  -->
<!--   limma.out = limma.out_,  -->
<!--   .converted.out = NULL,  -->
<!--   .readData.out = readData.out_,  -->
<!--   .input_noIDConversion = TRUE,  -->
<!--   input_dataFileFormat = 1,  -->
<!--   input_CountsDEGMethod = 3,  -->
<!--   input_selectModelComprions = paste0(colnames(readSampleInfo.out_)[1], ": ", -->
<!--                                       unique(readSampleInfo.out_[,1])[1], " vs. ", -->
<!--                                       unique(readSampleInfo.out_[,1])[2]), #"Gravity: Microgravity vs. Terrestrial", # format as "group: control vs. mutant" -->
<!--   input_selectFactorsModel = colnames(readSampleInfo.out_)[1], #"Gravity",  -->
<!--   factorReferenceLevels.out = paste(colnames(readSampleInfo.out_)[1], -->
<!--                                     unique(readSampleInfo.out_[,1])[2], sep = ":"), #"Gravity:Terrestrial" -->
<!--   input_selectContrast = paste(unique(readSampleInfo.out_[,1])[1],  -->
<!--                                unique(readSampleInfo.out_[,1])[2], -->
<!--                                sep = "-") # "Microgravity-Terrestrial" -->
<!-- ) -->

<!-- selectedHeatmap(selectedHeatmap.data.out = selectedHeatmap.data.out_,  -->
<!--                 .mycolors = mycolors_,  -->
<!--                 .heatColors = heatColors_) # heatmap for DEGs in selected comparison -->

<!-- # Save gene lists and data into files -->
<!-- write.csv( -->
<!--   selectedHeatmap.data(convertedData.out = convertedData.out_,  -->
<!--                        readSampleInfo.out = readSampleInfo.out_, -->
<!--                        .converted.out = converted.out_,  -->
<!--                        .readData.out = readData.out_,  -->
<!--                        .input_noIDConversion = input_noIDConversion_,  -->
<!--                        input_dataFileFormat = 1,  -->
<!--                        input_CountsDEGMethod = 3,  -->
<!--                        input_selectModelComprions = paste0(colnames(readSampleInfo.out_)[1], ": ", -->
<!--                                                            unique(readSampleInfo.out_[,1])[1], " vs. ", -->
<!--                                                            unique(readSampleInfo.out_[,1])[2]), #"Gravity: Microgravity vs. Terrestrial", # format as "group: control vs. mutant" -->
<!--                        input_selectFactorsModel = colnames(readSampleInfo.out_)[1], #"Gravity",  -->
<!--                        factorReferenceLevels.out = paste(colnames(readSampleInfo.out_)[1], -->
<!--                                                          unique(readSampleInfo.out_[,1])[2], sep = ":"), #"Gravity:Terrestrial" -->
<!--                        input_selectContrast = paste(unique(readSampleInfo.out_[,1])[1],  -->
<!--                                                     unique(readSampleInfo.out_[,1])[2], -->
<!--                                                     sep = "-"), -->
<!--                        limma.out = limma.out_)$genes,  -->
<!--   paste(params$input_folder, "heatmap.data.csv", sep = "/") -->
<!-- ) -->

<!-- write.csv(DEG.data(limma.out = limma.out_, convertedData.out = convertedData.out_, allGeneInfo.out = allGeneInfo.out_),  -->
<!--           paste(params$input_folder, "DEG.data.csv", sep = "/")) -->

<!-- write(AllGeneListsGMT(limma.out_),  -->
<!--       paste(params$input_folder, "AllGeneListsGMT.gmt", sep = "/")) -->
<!-- ```      -->
<!-- ```{r, message=FALSE  }  -->
<!-- input_selectGO2_ <- "GOBP" # Gene set category -->
<!-- geneListData.out_ <- geneListData(limma.out = limma.out_,  -->
<!--                                   allGeneInfo.out = allGeneInfo.out_,  -->
<!--                                   input_selectGO2 = "GOBP",  -->
<!--                                   input_selectOrg = "NEW",  -->
<!--                                   input_limmaPval = 0.1,  -->
<!--                                   input_limmaFC = 2,  -->
<!--                                   input_selectContrast = paste(unique(readSampleInfo.out_[,1])[1],  -->
<!--                                                                unique(readSampleInfo.out_[,1])[2], -->
<!--                                                                sep = "-")# "Microgravity-Terrestrial" -->
<!-- ) -->

<!-- volcanoPlot(limma.out = limma.out_,  -->
<!--             input_limmaPval = 0.1,  -->
<!--             input_limmaFC = 2,  -->
<!--             input_selectContrast = paste(unique(readSampleInfo.out_[,1])[1],  -->
<!--                                          unique(readSampleInfo.out_[,1])[2], -->
<!--                                          sep = "-")# "Microgravity-Terrestrial" -->
<!-- ) -->
<!-- ```      -->
<!-- ```{r, message=FALSE  }   -->
<!-- scatterPlot(limma.out = limma.out_,  -->
<!--             convertedData.out = convertedData.out_,  -->
<!--             readSampleInfo.out = readSampleInfo.out_,  -->
<!--             input_dataFileFormat = 1, -->
<!--             input_CountsDEGMethod = 3,  -->
<!--             input_limmaPval = 0.1,  -->
<!--             input_limmaFC = 2,  -->
<!--             input_selectModelComprions = paste0(colnames(readSampleInfo.out_)[1], ": ", -->
<!--                                                 unique(readSampleInfo.out_[,1])[1], " vs. ", -->
<!--                                                 unique(readSampleInfo.out_[,1])[2]), #"Gravity: Microgravity vs. Terrestrial", # format as "group: control vs. mutant" -->
<!--             input_selectFactorsModel = colnames(readSampleInfo.out_)[1], #"Gravity",  -->
<!--             factorReferenceLevels.out = paste(colnames(readSampleInfo.out_)[1], -->
<!--                                               unique(readSampleInfo.out_[,1])[2], sep = ":"), #"Gravity:Terrestrial" -->
<!--             input_selectContrast = paste(unique(readSampleInfo.out_[,1])[1],  -->
<!--                                          unique(readSampleInfo.out_[,1])[2], -->
<!--                                          sep = "-")) -->

<!-- ``` -->

<!-- ```{r, message=FALSE  }  -->
<!-- MAplot( -->
<!--   limma.out = limma.out_,  -->
<!--   convertedData.out = convertedData.out_,  -->
<!--   readSampleInfo.out = readSampleInfo.out_,  -->
<!--   .converted.out = converted.out_,  -->
<!--   .readData.out = readData.out_,  -->
<!--   .input_noIDConversion = TRUE,  -->
<!--   input_dataFileFormat = 1,  -->
<!--   input_CountsDEGMethod = 3,  -->
<!--   input_limmaPval = 0.1,  -->
<!--   input_limmaFC = 2,  -->
<!--   input_selectModelComprions = paste0(colnames(readSampleInfo.out_)[1], ": ", -->
<!--                                       unique(readSampleInfo.out_[,1])[1], " vs. ", -->
<!--                                       unique(readSampleInfo.out_[,1])[2]), #"Gravity: Microgravity vs. Terrestrial", # format as "group: control vs. mutant" -->
<!--   input_selectFactorsModel = colnames(readSampleInfo.out_)[1], #"Gravity",  -->
<!--   factorReferenceLevels.out = paste(colnames(readSampleInfo.out_)[1], -->
<!--                                     unique(readSampleInfo.out_[,1])[2], sep = ":"), #"Gravity:Terrestrial" -->
<!--   input_selectContrast = paste(unique(readSampleInfo.out_[,1])[1],  -->
<!--                                unique(readSampleInfo.out_[,1])[2], -->
<!--                                sep = "-") -->
<!-- ) -->
<!-- ```      -->
<!-- ```{r, message=FALSE}   -->
<!-- geneListGOTable.out_ <- geneListGOTable(GeneSets.out = GeneSets.out_,  -->
<!--                                         minGenesEnrichment = 2,  -->
<!--                                         selectedHeatmap.data.out = selectedHeatmap.data.out_) -->
<!-- # Read pathway data again -->
<!-- GeneSets.out_ <- readGeneSets( -->
<!--   fileName = geneSetFile_, -->
<!--   convertedData = convertedData.out_,  -->
<!--   GO = "GOBP",  -->
<!--   selectOrg = "NEW", -->
<!--   myrange = c(15, 2000) -->
<!-- ) -->

<!-- #input_removeRedudantSets_ <- TRUE # Remove highly redundant gene sets? -->
<!-- results <- geneListGO(geneListGOTable.out = geneListGOTable.out_,  -->
<!--                       input_removeRedudantSets = TRUE) # Enrichment analysis -->

<!-- results$adj.Pval <- format(results$adj.Pval, digits = 3) -->

<!-- kable(results, row.names = FALSE) %>% -->
<!--   kable_styling(bootstrap_options = c("striped", "hover")) %>% -->
<!--   scroll_box(width = "100%") -->
<!-- ``` -->

<!-- STRING-db API access.  -->
<!-- We need to find the taxonomy id of your species, this used by STRING. -->
<!-- First we try to guess the ID based on iDEP's database. Users can also skip this step and assign NCBI taxonomy id directly by -->
<!-- findTaxonomyID.out_ = 10090 # mouse 10090, human 9606 etc. -->

<!-- ```{r, message = FALSE} -->
<!-- STRING10_species_ <- read.csv(STRING10_speciesFile_) -->
<!-- ix <- grep("Arabidopsis thaliana", STRING10_species_$official_name) -->
<!-- findTaxonomyID.out_ <- STRING10_species_[ix, 1] # find taxonomyID -->
<!-- findTaxonomyID.out_ -->
<!-- ```  -->

<!-- Enrichment analysis using STRING      -->

<!-- ```{r, message=FALSE  }   -->
<!-- STRINGdb_geneList.out_ <- STRINGdb_geneList(geneListData.out = geneListData.out_,  -->
<!--                                             findTaxonomyID.out = findTaxonomyID.out_) # convert gene lists -->

<!-- # input_STRINGdbGO_ <- "Process" #' Process', 'Component', 'Function', 'KEGG', 'Pfam', 'InterPro' -->
<!-- results <- stringDB_GO_enrichmentData(selectedHeatmap.data.out = selectedHeatmap.data.out_,  -->
<!--                                       minGenesEnrichment = 2,  -->
<!--                                       input_STRINGdbGO = "Process", #' Process', 'Component', 'Function', 'KEGG', 'Pfam', 'InterPro' -->
<!--                                       findTaxonomyID.out = findTaxonomyID.out_,  -->
<!--                                       STRINGdb_geneList.out = STRINGdb_geneList.out_) # enrichment using STRING -->

<!-- results$adj.Pval <- format(results$adj.Pval, digits = 3) -->

<!-- kable(results, row.names = FALSE) %>% -->
<!--   kable_styling(bootstrap_options = c("striped", "hover")) %>% -->
<!--   scroll_box(width = "100%") -->
<!-- ```  -->

<!-- PPI network retrieval and analysis     -->

<!-- ```{r, message=FALSE  }  -->
<!-- stringDB_network1(geneLists = 1,  -->
<!--                   input_nGenesPPI = 100,  -->
<!--                   findTaxonomyID.out = findTaxonomyID.out_,  -->
<!--                   STRINGdb_geneList.out = STRINGdb_geneList.out_) # Show PPI network -->
<!-- ```  -->

<!-- Generating interactive PPI    -->
<!-- ```{r, message=FALSE  }   -->
<!-- write( -->
<!--   stringDB_network_link( -->
<!--     input_nGenesPPI = 100,  -->
<!--     findTaxonomyID.out = findTaxonomyID.out_,  -->
<!--     STRINGdb_geneList.out = STRINGdb_geneList.out_,  -->
<!--     geneListData.out = geneListData.out_ -->
<!--   ),  -->
<!--   paste(params$input_folder, "PPI_results.html", sep = "/") -->
<!-- ) -->
<!-- ``` -->

<!-- ## 8. Pathway analysis -->

<!-- ```{r, message=FALSE  }  -->
<!-- # Read pathway data again -->
<!-- GeneSets.out_ <- readGeneSets( -->
<!--   fileName = geneSetFile_, -->
<!--   convertedData = convertedData.out_,  -->
<!--   GO = "GOBP",  -->
<!--   selectOrg = "NEW", -->
<!--   myrange = c(15, 2000) -->
<!-- ) -->

<!-- gagePathwayData.out_ <- gagePathwayData(limma.out = limma.out_,  -->
<!--                                         input_minSetSize = 15,  -->
<!--                                         input_maxSetSize = 2000,  -->
<!--                                         input_selectContrast1 = paste(unique(readSampleInfo.out_[,1])[1],  -->
<!--                                                                       unique(readSampleInfo.out_[,1])[2], -->
<!--                                                                       sep = "-"), #"Microgravity-Terrestrial",  -->
<!--                                         input_pathwayPvalCutoff = 0.2,  -->
<!--                                         input_nPathwayShow = 30,  -->
<!--                                         input_absoluteFold = FALSE,  -->
<!--                                         input_GenePvalCutoff = 1,  -->
<!--                                         GeneSets.out = GeneSets.out_) # pathway analysis using GAGE -->

<!-- results <- gagePathwayData.out_ # Enrichment analysis for k-Means clusters -->
<!-- results$adj.Pval <- format(results$adj.Pval, digits = 3) -->
<!-- kable(results, row.names = FALSE) %>% -->
<!--   kable_styling(bootstrap_options = c("striped", "hover")) %>% -->
<!--   scroll_box(width = "100%") -->
<!-- ```      -->
<!-- ```{r, message=FALSE  }   -->
<!-- pathwayListData.out_ <- pathwayListData(allGeneInfo.out = allGeneInfo.out_,  -->
<!--                                         input_selectOrg = "NEW",  -->
<!--                                         input_selectGO = "GOBP",  -->
<!--                                         input_pathwayMethod = 1,  -->
<!--                                         gagePathwayData.out = gagePathwayData.out_,  -->
<!--                                         fgseaPathwayData.out = fgseaPathwayData.out_,  -->
<!--                                         GeneSets.out = GeneSets.out_) -->

<!-- enrichmentPlot(pathwayListData.out_, enrichedTerms = 25) -->
<!-- ```      -->
<!-- ```{r, message=FALSE  }   -->
<!-- enrichmentNetwork(pathwayListData.out_) -->
<!-- ```      -->

<!-- ```{r, message=FALSE  }   -->
<!-- enrichmentNetworkPlotly(pathwayListData.out_) -->
<!-- ```      -->

<!-- ```{r, message=FALSE  }   -->
<!-- # input_pathwayMethod_ <- 3 # 1  fgsea -->

<!-- fgseaPathwayData.out_ <- fgseaPathwayData(limma.out = limma.out_,  -->
<!--                                           input_minSetSize = 15,  -->
<!--                                           input_maxSetSize = 2000,  -->
<!--                                           input_selectContrast1 = paste(unique(readSampleInfo.out_[,1])[1],  -->
<!--                                                                         unique(readSampleInfo.out_[,1])[2], -->
<!--                                                                         sep = "-"), -->
<!--                                           input_pathwayPvalCutoff = 0.2,  -->
<!--                                           input_absoluteFold = FALSE,  -->
<!--                                           input_nPathwayShow = 30,  -->
<!--                                           input_GenePvalCutoff = 1,  -->
<!--                                           GeneSets.out = GeneSets.out_) # Pathway analysis using fgsea -->

<!-- results <- fgseaPathwayData.out_ # Enrichment analysis for k-Means clusters -->
<!-- results$adj.Pval <- format(results$adj.Pval, digits = 3) -->
<!-- kable(results, row.names = FALSE) %>% -->
<!--   kable_styling(bootstrap_options = c("striped", "hover")) %>% -->
<!--   scroll_box(width = "100%") -->
<!-- ```      -->
<!-- ```{r, message=FALSE  }   -->
<!-- pathwayListData.out_ <- pathwayListData(allGeneInfo.out = allGeneInfo.out_,  -->
<!--                                         input_selectOrg = "NEW",  -->
<!--                                         input_selectGO = "GOBP",  -->
<!--                                         input_pathwayMethod = 3,  -->
<!--                                         gagePathwayData.out = gagePathwayData.out_,  -->
<!--                                         fgseaPathwayData.out = fgseaPathwayData.out_,  -->
<!--                                         GeneSets.out = GeneSets.out_) -->

<!-- enrichmentPlot(enrichedTerms = pathwayListData.out_,  -->
<!--                rightMargin = 25,  -->
<!--                mycolors = mycolors_) -->
<!-- ```      -->
<!-- ```{r, message=FALSE  }   -->
<!-- enrichmentNetwork(pathwayListData.out_) -->
<!-- ``` -->

<!-- ```{r, message = FALSE} -->
<!-- enrichmentNetworkPlotly(pathwayListData.out_) -->
<!-- ```      -->
<!-- ```{r, message=FALSE,fig.width=9, fig.height=8  }   -->
<!-- PGSEAplot(convertedData.out = convertedData.out_,  -->
<!--           readSampleInfo.out = readSampleInfo.out_,  -->
<!--           input_selectOrg = "NEW",  -->
<!--           input_dataFileFormat = 1,  -->
<!--           input_selectGO = "GOBP",  -->
<!--           input_minSetSize = 15,  -->
<!--           input_maxSetSize = 2000,  -->
<!--           input_CountsDEGMethod = 3,  -->
<!--           input_selectModelComprions = paste0(colnames(readSampleInfo.out_)[1], ": ", -->
<!--                                               unique(readSampleInfo.out_[,1])[1], " vs. ", -->
<!--                                               unique(readSampleInfo.out_[,1])[2]), #"Gravity: Microgravity vs. Terrestrial", # format as "group: control vs. mutant" -->
<!--           input_selectFactorsModel = colnames(readSampleInfo.out_)[1], #"Gravity",  -->
<!--           factorReferenceLevels.out = paste(colnames(readSampleInfo.out_)[1], -->
<!--                                             unique(readSampleInfo.out_[,1])[2], sep = ":"), #"Gravity:Terrestrial" -->
<!--           input_selectContrast1 = paste(unique(readSampleInfo.out_[,1])[1],  -->
<!--                                         unique(readSampleInfo.out_[,1])[2], -->
<!--                                         sep = ":"), -->
<!--           input_pathwayPvalCutoff = 0.2,  -->
<!--           input_nPathwayShow = 30,  -->
<!--           GeneSets.out = GeneSets.out_) # pathway analysis using PGSEA -->
<!-- ``` -->

<!-- ## 9. Chromosome -->

<!-- ```{r, message=FALSE  }  -->
<!-- #input_selectContrast2_ <- "Microgravity-Terrestrial" # select Comparison -->
<!-- # input_selectContrast2 = limma.out_$comparisons[3] # manually set -->
<!-- #input_limmaPvalViz_ <- 0.1 # FDR to filter genes -->
<!-- #input_limmaFCViz_ <- 2 # FDR to filter genes -->
<!-- genomePlotly(limma.out = limma.out_,  -->
<!--              allGeneInfo.out = allGeneInfo.out_,  -->
<!--              input_selectContrast2 = paste(unique(readSampleInfo.out_[,1])[1],  -->
<!--                                            unique(readSampleInfo.out_[,1])[2], -->
<!--                                            sep = "-"), -->
<!--              input_limmaPvalViz = 0.1,  -->
<!--              input_limmaFCViz = 2) # shows fold-changes on the genome -->
<!-- ``` -->

<!-- ## 10. Biclustering -->

<!-- ```{r, message=FALSE  }  -->
<!-- biclustering.out_ <- biclustering(convertedData.out = convertedData.out_,  -->
<!--                                   input_nGenesBiclust = 1000,  -->
<!--                                   input_biclustMethod = "BCCC()") # run analysis -->

<!-- input_selectBicluster_ <- 1 # select a cluster -->
<!-- biclustHeatmap(biclustering.out = biclustering.out_,  -->
<!--                heatColors = heatColors_,  -->
<!--                input_heatColors1 = 1,  -->
<!--                input_selectBicluster = 1) # heatmap for selected cluster -->
<!-- ``` -->

<!-- ```{r, message=FALSE  }  -->
<!-- #input_selectGO4_ <- "GOBP" # Gene set category -->
<!-- # Read pathway data again -->
<!-- GeneSets.out_ <- readGeneSets( -->
<!--   fileName = geneSetFile_, -->
<!--   convertedData = convertedData.out_,  -->
<!--   GO = "GOBP",  -->
<!--   selectOrg = "NEW", -->
<!--   myrange = c(15, 2000) -->
<!-- ) -->

<!-- results <- geneListBclustGO(minGenesEnrichment = 2,  -->
<!--                             input_selectBicluster = 1,  -->
<!--                             biclustering.out = biclustering.out_,  -->
<!--                             GeneSets.out = GeneSets.out_) # Enrichment analysis for k-Means clusters -->

<!-- results$adj.Pval <- format(results$adj.Pval, digits = 3) -->

<!-- kable(results, row.names = FALSE) %>% -->
<!--   kable_styling(bootstrap_options = c("striped", "hover")) %>% -->
<!--   scroll_box(width = "100%") -->
<!-- ``` -->

<!-- ## 11. Co-expression network -->

<!-- ```{r, message=FALSE  }  -->
<!-- wgcna.out_ <- wgcna(convertedData.out = convertedData.out_,  -->
<!--                     maxGeneWGCNA = 2000,  -->
<!--                     input_mySoftPower = 5,  -->
<!--                     input_nGenesNetwork = 1000,  -->
<!--                     input_minModuleSize = 20) # run WGCNA -->
<!-- ``` -->

<!-- ```{r, message=FALSE  }   -->
<!-- softPower(wgcna.out_) # soft power curve -->
<!-- ```      -->

<!-- ```{r, message=FALSE  }   -->
<!-- listWGCNA.Modules.out <- listWGCNA.Modules(wgcna.out_) # modules -->
<!-- modulePlot(wgcna.out_) # plot modules -->
<!-- ```      -->

<!-- ```{r, message=FALSE  }  -->
<!-- # Read pathway data again -->
<!-- GeneSets.out_ <- readGeneSets( -->
<!--   fileName = geneSetFile_, -->
<!--   convertedData = convertedData.out_,  -->
<!--   GO = "GOBP",  -->
<!--   selectOrg = "NEW", -->
<!--   myrange = c(15, 2000) -->
<!-- ) -->

<!-- moduleNetwork(wgcna.out = wgcna.out_,  -->
<!--               input_noIDConversion = TRUE,  -->
<!--               allGeneInfo.out = allGeneInfo.out_,  -->
<!--               input_selectOrg = "NEW",  -->
<!--               input_selectWGCNA.Module = "Entire network",  -->
<!--               input_topGenesNetwork = 10,  -->
<!--               input_edgeThreshold = 0.4,  -->
<!--               input_selectGO5 = "GOBP") # show network of top genes in selected module -->
<!-- ```      -->
<!-- ```{r, message=FALSE  }  -->
<!-- results <- networkModuleGO(GeneSets.out = GeneSets.out_,  -->
<!--                            minGenesEnrichment = 2,  -->
<!--                            input_selectWGCNA.Module = "Entire network",  -->
<!--                            wgcna.out = wgcna.out_) # Enrichment analysis of selected module -->

<!-- results$adj.Pval <- format(results$adj.Pval, digits = 3) -->
<!-- kable(results, row.names = FALSE) %>% -->
<!--   kable_styling(bootstrap_options = c("striped", "hover")) %>% -->
<!--   scroll_box(width = "100%") -->
<!-- ```    -->
